scripts/earthengine/utils.py

```diff
@@ -649,3 +435,61 @@ def str_format_float(data: float, precision_digits: int = 6):
     if isinstance(data, float):
         return f'{data:.{precision_digits}f}'
     return data
+
+
+# Date utilities
+
+
+def date_today(date_format: str = '%Y-%m-%d') -> str:
+    '''Returns today's date in the given format.'''
+    return date.today().strftime(date_format)
+
+
+def date_yesterday(date_format: str = '%Y-%m-%d') -> str:
+    '''Returns today's date in the given format.'''
+    return date_advance_by_period(date_today(date_format), '-1d', date_format)
+
+
+def date_parse_time_period(time_period: str) -> (int, str):
+    '''Parse time period into a tuple of (number, unit), for eg: P1M: (1, month).'''
+    re_pat = r'P?(?P<delta>[+-]?[0-9]+)(?P<unit>[A-Z])'
+    m = re.search(re_pat, time_period.upper())
+    if m:
+        m_dict = m.groupdict()
+        delta = int(m_dict.get('delta', '0'))
+        unit = m_dict.get('unit', 'M')
+        period_dict = {'D': 'days', 'M': 'months', 'Y': 'years'}
+        period = period_dict.get(unit, 'day')
+        return (delta, period)
+    return (0, 'days')
+
+
+def date_advance_by_period(date_str: str,
+                           time_period: str,
+                           date_format: str = '%Y-%m-%d') -> str:
+    '''Returns the date string advanced by the time period.'''
+    next_date = ''
```

Comment:
nit: looks like `next_date` variable is only ever used to store the empty string. So perhaps wherever this function is returning `next_date`, it can just return the empty string ('') instead of `next_date`?

Reply:
removed next_date